### PR TITLE
DD4hep: DDFilteredView Cleanup and One More Test

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Remove legacy ```extractParameters``` member function: replaced by ```parameters```
* Add a special case to handle a Boolean shape parameters
* Consolidate ```nextSibling```. Find the first sibling of the node matching the filter and change an iteration direction to BFS when it's called first time, otherwise return the next matching sibling. Returns false when there are no matching siblings.
* Build history on demand
* Run the DT validation tests automatically

#### PR validation:

unit tests and relval tests

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

File: DetectorDescription/DDCMS/src/DDFilteredView.cc [hunk 1]
```diff
@@ -148,20 +148,22 @@ bool DDFilteredView::firstSibling() {
   assert(node_);
   if (it_.empty() || currentFilter_ == nullptr)
     return false;
+  Node* node = nullptr;
   if (next(0) == false)
     return false;
+  node = node_;
```
[the remark this answers is not in the document]
Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc]: done

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
@fabiocos - please, merge this one

**Comment by fabiocos** [comment]
+1

**Pull request merged**